llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -1908,15 +1908,9 @@ struct DSEState {
                               Malloc->getArgOperand(0), IRB, TLI);
     if (!Calloc)
       return false;
-    MemorySSAUpdater Updater(&MSSA);
-    auto *NewAccess =
-      Updater.createMemoryAccessAfter(cast<Instruction>(Calloc), nullptr,
-                                      MallocDef);
-    auto *NewAccessMD = cast<MemoryDef>(NewAccess);
-    Updater.insertDef(NewAccessMD, /*RenameUses=*/true);
```

Comment:
Don't we have to keep the part of this that creates the new access?

Reply:
Ah yes added back. Strangely this doesn't seem to be caught be the verifier in any of the tests we have.